// react-liquidswipe/dist/react-liquidswipe.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var web = require('@react-spring/web');
var reactUseGesture = require('react-use-gesture');
var styled = _interopDefault(require('styled-components'));

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

var _templateObject, _templateObject2, _templateObject3;
var LQSW_SVG = /*#__PURE__*/styled.svg(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    position: absolute;\n    height: 100%;\n    width: 10px;\n"])));
var LQSW_PAGE = /*#__PURE__*/styled.div(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    width: 100%;\n    height: 100%;\n    position: absolute;\n"])));
var LQSW_CONTAINER = /*#__PURE__*/styled.div(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n    width: 100vw;\n    height: 100vh;\n    position: relative;\n    overflow: hidden;\n    @media only screen and (max-width: 500px) {\n      width: 100%;\n    }\n"])));
var LQSW_BUTTON = {
  position: "absolute",
  width: "40px",
  height: "40px",
  background: "transparent",
  cursor: "pointer",
  border: "none",
  touchAction: "pan-y",
  outline: 0
};

// const width = window.innerWidth;
// let w = width;

var getPath = function getPath(y, x, w, h) {
  var anchorDistance = 200 + x * 0.5;
  var curviness = anchorDistance - 80;
  var path = "M" + w + ", " + h + " H0V0h" + w + "v " + (y - anchorDistance) + " c0, " + curviness + " , " + x + " , " + curviness + " , " + x + " , " + anchorDistance + " \n  S" + w + ", " + y + " ," + w + ", " + (y + anchorDistance * 2) + " V " + h + " z";
  return path;
};

function Page(_ref) {
  var prev = _ref.prev,
      current = _ref.current,
      next = _ref.next,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? "#333333" : _ref$theme,
      index = _ref.index,
      setActive = _ref.setActive,
      _ref$gone = _ref.gone,
      gone = _ref$gone === void 0 ? false : _ref$gone,
      parentElement = _ref.parentElement;
  var height = React.useState(parentElement.current.offsetHeight)[0];
  var width = React.useState(parentElement.current.offsetWidth)[0];

  var _useState = React.useState(gone),
      isGone = _useState[0],
      setGone = _useState[1];

  var _useState2 = React.useState(false),
      isPrevMove = _useState2[0],
      setPrevMove = _useState2[1];

  var _useState3 = React.useState(false),
      isNextMove = _useState3[0],
      setNextMove = _useState3[1];

  var _useSpring = web.useSpring(function () {
    return {
      posX: -50,
      posY: height * 0.72 - 20,
      config: {
        mass: 3
      }
    };
  }),
      posL = _useSpring[0],
      setPosL = _useSpring[1];

  var _useSpring2 = web.useSpring(function () {
    return {
      posX: width + 50,
      posY: height * 0.72 - 20,
      config: {
        mass: 3
      }
    };
  }),
      posR = _useSpring2[0],
      setPosR = _useSpring2[1];

  var _useSpring3 = web.useSpring(function () {
    return {
      prevD: gone ? getPath(0, 0, width, height) : getPath(height * 0.72, 0, 0, height),
      config: {
        mass: 3
      },
      onRest: function onRest() {
        if (isGone) {
          setDPrevValue(getPath(0, 0, width, height));
        }
      }
    };
  }),
      prevD = _useSpring3[0].prevD,
      setDPrevValue = _useSpring3[1];

  var _useSpring4 = web.useSpring(function () {
    return {
      nextD: gone ? getPath(0, 0, width, height) : getPath(height * 0.72, 0, 0, height),
      config: {
        mass: 3
      },
      onRest: function onRest() {
        if (isGone) {
          setDNextValue(getPath(0, 0, width, height));
        }
      }
    };
  }),
      nextD = _useSpring4[0].nextD,
      setDNextValue = _useSpring4[1];

  React.useEffect(function () {
    if (!gone) {
      setDPrevValue({
        prevD: getPath(height * 0.72, 48, 5, height)
      });
      setDNextValue({
        nextD: getPath(height * 0.72, 48, 5, height)
      });
      setTimeout(function () {
        setPosL({
          posX: 7
        });
        setPosR({
          posX: width - 47
        });
      }, 100);
    }
  }, [gone, setDPrevValue, setDNextValue, setPosL, setPosR]);
  var bind = reactUseGesture.useDrag(function (_ref2) {
    var down = _ref2.down,
        _ref2$movement = _ref2.movement,
        mx = _ref2$movement[0],
        _ref2$xy = _ref2.xy,
        my = _ref2$xy[1],
        _ref2$vxvy = _ref2.vxvy,
        vx = _ref2$vxvy[0];

    if (!isGone) {
      if (down && isPrevMove) {
        setDPrevValue({
          prevD: getPath(my, mx + 60, 10, height)
        });
        setPosL({
          posX: mx + 20,
          posY: my - 20
        });

        if (mx > width / 2 || vx > 3) {
          setDPrevValue({
            prevD: getPath(my, -50, width, height)
          });
          setGone(true);
          setTimeout(function () {
            setDPrevValue({
              prevD: getPath(my, 0, width, height)
            });
            setActive(index - 1);
          }, 240);
        }
      } else if (down && isNextMove) {
        mx = mx * -1;
        setDNextValue({
          nextD: getPath(my, mx + 60, 10, height)
        });
        setPosR({
          posX: width - (mx + 60),
          posY: my - 20
        });

        if (Math.abs(mx) > width / 2 || vx < -3) {
          setDNextValue({
            nextD: getPath(my, -50, width, height)
          });
          setGone(true);
          setTimeout(function () {
            setDNextValue({
              nextD: getPath(my, 0, width, height)
            });
            setActive(index + 1);
          }, 240);
        }
      } else {
        setDPrevValue({
          prevD: getPath(height * 0.72, 48, 5, height)
        });
        setPosL({
          posX: 7,
          posY: height * 0.72 - 20
        });
        setDNextValue({
          nextD: getPath(height * 0.72, 48, 5, height)
        });
        setPosR({
          posX: width - 47,
          posY: height * 0.72 - 20
        });
      }
    }
  }, {
    useTouch: width <= 414
  });
  return React__default.createElement("div", Object.assign({
    id: "pageContainer" + index
  }, bind()), React__default.createElement(LQSW_SVG, {
    style: {
      zIndex: 20004
    },
    version: "1.1",
    id: "blob",
    xmlns: "http://www.w3.org/2000/svg"
  }, React__default.createElement("clipPath", {
    id: "clipping" + index
  }, React__default.createElement(web.animated.path, {
    id: "blob-path" + index,
    d: prevD
  }))), React__default.createElement(LQSW_SVG, {
    style: {
      zIndex: 20004
    },
    version: "1.1",
    id: "blob",
    xmlns: "http://www.w3.org/2000/svg"
  }, React__default.createElement("clipPath", {
    id: "clippingRight" + index
  }, React__default.createElement(web.animated.path, {
    transform: "translate(" + width + ", 0) scale(-1, 1)",
    id: "blob-path" + index,
    d: nextD
  }))), prev && React__default.createElement(React__default.Fragment, null, React__default.createElement(LQSW_PAGE, {
    style: {
      clipPath: "url(#clipping" + index + ")",
      WebkitClipPath: "url(#clipping" + index + ")",
      zIndex: 20001
    }
  }, prev), React__default.createElement(web.animated.button, {
    id: "button1i" + index,
    color: theme,
    onMouseDown: function onMouseDown() {
      setPrevMove(true);
    },
    onMouseUp: function onMouseUp() {
      setPrevMove(false);
    },
    onTouchStart: function onTouchStart() {
      setPrevMove(true);
    },
    onTouchEnd: function onTouchEnd() {
      setPrevMove(false);
    },
    // @ts-ignore
    style: _extends({}, LQSW_BUTTON, {
      opacity: posL.posX.interpolate({
        range: [0, 100],
        output: [1, 0]
      }),
      zIndex: 20005,
      transform: web.interpolate([posL.posX.interpolate(function (x) {
        return "translateX(" + x + "px)";
      }), posL.posY.interpolate(function (y) {
        return "translateY(" + y + "px)";
      })], function (translateX, translateY) {
        return translateX + " " + translateY;
      })
    })
  }, React__default.createElement("svg", {
    width: "30",
    height: "30",
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, React__default.createElement("path", {
    d: "M16.6308 13.131C16.5743 13.189 16.3609 13.437 16.1622 13.641C14.9971 14.924 11.9576 17.024 10.3668 17.665C10.1252 17.768 9.51437 17.986 9.18802 18C8.8753 18 8.5772 17.928 8.29274 17.782C7.93814 17.578 7.65368 17.257 7.49781 16.878C7.39747 16.615 7.2416 15.828 7.2416 15.814C7.08573 14.953 7 13.554 7 12.008C7 10.535 7.08573 9.193 7.21335 8.319C7.22796 8.305 7.38383 7.327 7.55431 6.992C7.86702 6.38 8.47784 6 9.13151 6H9.18802C9.61374 6.015 10.509 6.395 10.509 6.409C12.0141 7.051 14.9834 9.048 16.1768 10.375C16.1768 10.375 16.5129 10.716 16.659 10.929C16.887 11.235 17 11.614 17 11.993C17 12.416 16.8724 12.81 16.6308 13.131Z",
    fill: "#130F26"
  })))), React__default.createElement(LQSW_PAGE, {
    style: {
      zIndex: 20000
    }
  }, React__default.createElement(React__default.Fragment, null, current)), next && React__default.createElement(React__default.Fragment, null, React__default.createElement(LQSW_PAGE, {
    style: {
      clipPath: "url(#clippingRight" + index + ")",
      WebkitClipPath: "url(#clippingRight" + index + ")",
      zIndex: 20002
    }
  }, next), React__default.createElement(web.animated.button, {
    id: "button2i" + index,
    color: theme,
    onMouseDown: function onMouseDown() {
      setNextMove(true);
    },
    onMouseUp: function onMouseUp() {
      setNextMove(false);
    },
    onTouchStart: function onTouchStart() {
      setNextMove(true);
    },
    onTouchEnd: function onTouchEnd() {
      setNextMove(false);
    },
    // @ts-ignore
    style: _extends({}, LQSW_BUTTON, {
      opacity: posR.posX.interpolate({
        range: [width - 40, width - 140],
        output: [1, 0]
      }),
      zIndex: 20005,
      transform: web.interpolate([posR.posX.interpolate(function (x) {
        return "translateX(" + x + "px)";
      }), posR.posY.interpolate(function (y) {
        return "translateY(" + y + "px)";
      })], function (translateX, translateY) {
        return translateX + " " + translateY;
      })
    })
  }, React__default.createElement("svg", {
    width: "30",
    height: "30",
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, React__default.createElement("path", {
    d: "M7.36922 10.869C7.42572 10.811 7.63906 10.563 7.8378 10.359C9.00292 9.076 12.0424 6.976 13.6332 6.335C13.8748 6.232 14.4856 6.014 14.812 6C15.1247 6 15.4228 6.072 15.7073 6.218C16.0619 6.422 16.3463 6.743 16.5022 7.122C16.6025 7.385 16.7584 8.172 16.7584 8.186C16.9143 9.047 17 10.446 17 11.992C17 13.465 16.9143 14.807 16.7867 15.681C16.772 15.695 16.6162 16.673 16.4457 17.008C16.133 17.62 15.5222 18 14.8685 18H14.812C14.3863 17.985 13.491 17.605 13.491 17.591C11.9859 16.949 9.01656 14.952 7.82319 13.625C7.82319 13.625 7.48709 13.284 7.34096 13.071C7.11301 12.765 7 12.386 7 12.007C7 11.584 7.12762 11.19 7.36922 10.869Z",
    fill: "#130F26"
  })))));
}

function LiquidSwipe(_ref) {
  var components = _ref.components,
      style = _ref.style;
  var sizeOfSwipe = components.length;

  var _useState = React.useState(0),
      isActive = _useState[0],
      setActive = _useState[1];

  var _useState2 = React.useState(),
      elm = _useState2[0],
      setElm = _useState2[1];

  var parentElement = React.useRef(null);
  React.useEffect(function () {
    setElm(React__default.createElement(Page, {
      key: isActive,
      index: isActive,
      prev: components[isActive - 1] || null,
      current: components[isActive],
      next: components[isActive + 1] || null,
      setActive: setActive,
      parentElement: parentElement
    }));
  }, [isActive, components]);

  if (sizeOfSwipe !== 0) {
    return React__default.createElement(React__default.Fragment, null, React__default.createElement(LQSW_CONTAINER, {
      ref: parentElement,
      style: style
    }, elm));
  } else {
    return React__default.createElement(React__default.Fragment, null);
  }
}

// import React, { FC, HTMLAttributes, ReactChild } from 'react';

exports.default = LiquidSwipe;
