// react-seo-component/dist/react-seo-component.cjs.development.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var reactHelmet = require('react-helmet');

var Facebook = function Facebook(_ref) {
  var url = _ref.url,
      type = _ref.type,
      title = _ref.title,
      desc = _ref.desc,
      image = _ref.image,
      locale = _ref.locale;
  return React.createElement(reactHelmet.Helmet, null, React.createElement("meta", {
    property: "og:locale",
    content: locale
  }), React.createElement("meta", {
    property: "og:url",
    content: url
  }), React.createElement("meta", {
    property: "og:type",
    content: type
  }), React.createElement("meta", {
    property: "og:title",
    content: title
  }), React.createElement("meta", {
    property: "og:description",
    content: desc
  }), React.createElement("meta", {
    property: "og:image",
    content: image
  }), React.createElement("meta", {
    property: "og:image:alt",
    content: desc
  }));
};

var Twitter = function Twitter(_ref) {
  var _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'summary_large_image' : _ref$type,
      username = _ref.username,
      title = _ref.title,
      desc = _ref.desc,
      image = _ref.image;
  return React.createElement(reactHelmet.Helmet, null, username && React.createElement("meta", {
    name: "twitter:creator",
    content: username
  }), React.createElement("meta", {
    name: "twitter:card",
    content: type
  }), React.createElement("meta", {
    name: "twitter:title",
    content: title
  }), React.createElement("meta", {
    name: "twitter:description",
    content: desc
  }), React.createElement("meta", {
    name: "twitter:image",
    content: image
  }), React.createElement("meta", {
    name: "twitter:image:alt",
    content: desc
  }));
};

var SEO = function SEO(_ref) {
  var title = _ref.title,
      titleTemplate = _ref.titleTemplate,
      titleSeparator = _ref.titleSeparator,
      description = _ref.description,
      pathname = _ref.pathname,
      _ref$article = _ref.article,
      article = _ref$article === void 0 ? false : _ref$article,
      image = _ref.image,
      siteLanguage = _ref.siteLanguage,
      siteLocale = _ref.siteLocale,
      twitterUsername = _ref.twitterUsername,
      _ref$author = _ref.author,
      author = _ref$author === void 0 ? 'J Doe.' : _ref$author,
      datePublished = _ref.datePublished,
      dateModified = _ref.dateModified;
  var seo = {
    title: title.slice(0, 70),
    description: description.slice(0, 160),
    datePublished: datePublished ? null : new Date(Date.now()).toISOString(),
    dateModified: dateModified ? null : new Date(Date.now()).toISOString()
  };
  var copyrightYear = new Date().getFullYear(); // schema.org in JSONLD format
  // https://developers.google.com/search/docs/guides/intro-structured-data
  // You can fill out the 'author', 'creator' with more data or another type (e.g. 'Organization')
  // Structured Data Testing Tool >>
  // https://search.google.com/structured-data/testing-tool

  var schemaOrgWebPage = {
    '@context': 'http://schema.org',
    '@type': 'WebPage',
    url: pathname,
    headline: seo.description,
    inLanguage: siteLanguage,
    mainEntityOfPage: pathname,
    description: seo.description,
    name: seo.title,
    author: {
      '@type': 'Person',
      name: author
    },
    copyrightHolder: {
      '@type': 'Person',
      name: author
    },
    copyrightYear: copyrightYear,
    creator: {
      '@type': 'Person',
      name: author
    },
    publisher: {
      '@type': 'Person',
      name: author
    },
    datePublished: seo.datePublished,
    dateModified: seo.dateModified,
    image: {
      '@type': 'ImageObject',
      url: "" + image
    }
  }; // Initial breadcrumb list

  var itemListElement = [{
    '@type': 'ListItem',
    item: {
      '@id': pathname,
      name: 'Homepage'
    },
    position: 1
  }];
  var schemaArticle = null;

  if (article) {
    schemaArticle = {
      '@context': 'http://schema.org',
      '@type': 'Article',
      author: {
        '@type': 'Person',
        name: author
      },
      copyrightHolder: {
        '@type': 'Person',
        name: author
      },
      copyrightYear: copyrightYear,
      creator: {
        '@type': 'Person',
        name: author
      },
      publisher: {
        '@type': 'Organization',
        name: author,
        logo: {
          '@type': 'ImageObject',
          url: "" + image
        }
      },
      datePublished: seo.datePublished,
      dateModified: seo.dateModified,
      description: seo.description,
      headline: seo.title,
      inLanguage: siteLanguage,
      url: pathname,
      name: seo.title,
      image: {
        '@type': 'ImageObject',
        url: image
      },
      mainEntityOfPage: pathname
    }; // Push current blog post into breadcrumb list

    itemListElement.push({
      '@type': 'ListItem',
      item: {
        '@id': pathname,
        name: seo.title
      },
      position: 2
    });
  }

  var breadcrumb = {
    '@context': 'http://schema.org',
    '@type': 'BreadcrumbList',
    description: 'Breadcrumbs list',
    name: 'Breadcrumbs',
    itemListElement: itemListElement
  };
  return React.createElement(React.Fragment, null, React.createElement(reactHelmet.Helmet, {
    title: seo.title,
    titleTemplate: "%s " + (titleSeparator ? titleSeparator : "\xB7") + " " + titleTemplate
  }, React.createElement("html", {
    lang: siteLanguage ? siteLanguage : 'en'
  }), React.createElement("link", {
    rel: "canonical",
    href: pathname
  }), React.createElement("meta", {
    name: "description",
    content: seo.description
  }), !article && React.createElement("script", {
    type: "application/ld+json"
  }, JSON.stringify(schemaOrgWebPage)), article && React.createElement("script", {
    type: "application/ld+json"
  }, JSON.stringify(schemaArticle)), React.createElement("script", {
    type: "application/ld+json"
  }, JSON.stringify(breadcrumb))), image && React.createElement(React.Fragment, null, React.createElement(Facebook, {
    desc: seo.description,
    image: image,
    title: seo.title,
    type: article ? 'article' : 'website',
    url: pathname,
    locale: siteLocale ? siteLocale : 'en_gb'
  }), React.createElement(Twitter, {
    title: seo.title,
    image: image,
    desc: seo.description,
    username: twitterUsername
  })));
};

exports.Facebook = Facebook;
exports.Twitter = Twitter;
exports.default = SEO;
